core/include/mm/sp_mem.h

```diff
@@ -0,0 +1,80 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021, Arm Limited.
+ */
+
+#ifndef __MM_SP_MEM_H
+#define __MM_SP_MEM_H
+
+#include <ffa.h>
+#include <mm/fobj.h>
+#include <sys/queue.h>
+
+struct sp_mem_receiver;
+struct sp_mem;
+
+/*
+ * The sp_mem_receiver keeps track of the data which is specific to the
+ * receiving endpoint. When sharing memory via FF-A, each FF-A endpoint can
+ * have different permissions. Each of the endpoints can retrieve the share
+ * multiple times. The ref_count is used to track the amount of time the
+ * endpoint has retrieved the share.
+ */
+struct sp_mem_receiver {
+	struct ffa_mem_access_perm perm;
+	/* Keep track of the number of times the share has been retrieved. */
```

Comment:
can remove this since the description above.

IMO, prefer description like,
```
 * struct sp_mem_receiver - Memory object instance for a receiver endpoint
 * 
 * @perm: Receiver permission on the shared memory
 * @ref_count: Count retrieve requests from endpoint
 * @smem: Shared memory reference
 * @link: Link in related list

Reply:
Done